core/src/main/scala/org/apache/spark/JobArtifactSet.scala

```diff
@@ -33,13 +33,16 @@ import java.util.Objects
  * @param files Files belonging to this session.
  * @param archives Archives belonging to this session.
  */
-class JobArtifactSet(
+private[spark] class JobArtifactSet(
     val uuid: Option[String],
     val replClassDirUri: Option[String],
     val jars: Map[String, Long],
     val files: Map[String, Long],
     val archives: Map[String, Long]) extends Serializable {
-  def withActive[T](f: => T): T = JobArtifactSet.withActive(this)(f)
+  def withActive[T](f: => T): T = {
+    assert(uuid.nonEmpty)
```

Comment:
Adding to the comment regd `withActive`, if we make the constructor private and control the contents of the jars/files/archives here, we could drop this requirement